### PR TITLE
{chem}[foss/2019a,intel/2019a] RDKit v2019.09.1 w/ Python 3.7.2

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/3e741f181ee9dad3037c0823ebf3a27e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3169.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/ef9fd4a8f7d494d174dc36fe25cb567d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2404.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9e166ce1b78ca1595dd062c0fadb18c2 for a full test report.

File: easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    # see https://www.inchi-trust.org/downloads/
+    'https://www.inchi-trust.org/download/105/',
+    'https://sourceforge.net/projects/avalontoolkit/files/AvalonToolkit_1.2/',
+]
+sources = [
+    'Release_%s.tar.gz' % version.replace('.', '_'),
+    'v%s.tar.gz' % local_ringdecomposerlib_ver,
+    # add datestamp to unversioned INCHI-1-SRC.zip based on date of most recent modified file in zipball
+    {'download_filename': 'INCHI-1-SRC.zip', 'filename': 'INCHI-1-SRC-20170203.zip'},
+    {'filename': 'AvalonToolkit_1.2.0.source.tar', 'extract_cmd': "cp %s ."},
```
Comment (akesandgren) [inline easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb]: Should probably have a local_avalon_ver here and reuse that in the URL specification

File: easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+# point to location of dependencies (to prevent that they're auto-downloaded)
+configopts += "-DCATCH_DIR=$EBROOTCATCH2/include/catch2 "
+configopts += "-DMAEPARSER_DIR=$EBROOTMAEPARSER -DCOORDGEN_DIR=$EBROOTCOORDGENLIBS "
+
+# using empty value for -DINCHI_URL prevents auto-downloading
+configopts += "-DRDK_BUILD_CAIRO_SUPPORT=ON -DRDK_BUILD_INCHI_SUPPORT=ON -DINCHI_URL='' "
+
+# prevent auto-download of AvalonToolkit source tarball by using empty value for -DAVALONTOOLS_URL
+configopts += "-DRDK_BUILD_AVALON_SUPPORT=ON -DAVALONTOOLS_URL='file://%(builddir)s/AvalonToolkit_1.2.0.source.tar' "
```
Comment (akesandgren) [inline easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb]: I.e. resuse local_avalon_ver here

File: easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+description = "RDKit is a collection of cheminformatics and machine-learning software written in C++ and Python."
+
+toolchain = {'name': 'foss', 'version': '2019a'}
+
+source_urls = [
+    'https://github.com/rdkit/rdkit/archive/',
+    'https://github.com/rareylab/RingDecomposerLib/archive/',
+    # see https://www.inchi-trust.org/downloads/
+    'https://www.inchi-trust.org/download/105/',
+    'https://sourceforge.net/projects/avalontoolkit/files/AvalonToolkit_1.2/',
```
Comment (akesandgren) [inline easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb]: potential reuse of a local_avalon_ver here too.

File: easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# move unpacked INCHI-1-SRC to External/INCHI-API/src, where it is expected
+preconfigopts += "cd %(builddir)s/rdkit-*/External/INCHI-API/ && mv %(builddir)s/INCHI-1-SRC src && cd - && "
+
+# RDKit requires access to the sources of RingDecomposerLib
+# we download it ourselves and point to the unpacked source tarball to control the used version
+configopts = "-DURFLIB_DIR=%%(builddir)s/RingDecomposerLib-%s/src/RingDecomposerLib " % local_ringdecomposerlib_ver
+
+configopts += "-DPy_ENABLE_SHARED=1 -DRDK_INSTALL_INTREE=ON -DRDK_INSTALL_STATIC_LIBS=OFF -DRDK_INSTALL_INTREE=OFF "
+configopts += "-DBoost_INCLUDE_DIR=$EBROOTBOOST/include -DBoost_LIBRARY_DIR_RELEASE=$EBROOTBOOST/lib "
+configopts += "-DPYTHON_NUMPY_INCLUDE_PATH=$EBROOTPYTHON/lib/python*/site-packages/numpy-*/numpy/core/include "
```
Comment (akesandgren) [inline easybuild/easyconfigs/r/RDKit/RDKit-2019.09.1-foss-2019a-Python-3.7.2.eb]: Is this one really required? the numpy include path is part of CPATH, which gets injected into CMAKE_INCLUDE_DIRS

And didn't we have a generic way of telling where that is?

**Comment by verdurin** [comment]
@boegel Could you address @akesandgren 's comments?

**Comment by Darkless012** [comment]
@verdurin @akesandgren 
Requested changes are obsolete, please close or merge this.
There is already RDKit 2019.09.3 on 2019b toolchain merged with different install steps. https://github.com/easybuilders/easybuild-easyconfigs/pull/10063
There is also 2020a version pending https://github.com/easybuilders/easybuild-easyconfigs/pull/11013

**Pull request closed without merging**